### ISSUE TITLE
Top Bar overlapping with desktop icons

### ISSUE THREAD

**Issue by angularorbit** [mlutfy/hidetopbar#190]
Similar to how auto-hiding the dock makes it cover desktop icons, Hide Top Bar covers desktop icons. I would like the icons to default to positions not covered by the top bar. Also, I've noticed that it moves the dock upwards when activated, which I don't mind.

![](https://i.stack.imgur.com/bh97S.jpg)
Steps to reproduce:

Write this in a terminal while observing the desktop
```
gnome-shell-extension-tool -e gnomehidetopbar@mathieu.bidon.ca
gnome-shell-extension-tool -d gnomehidetopbar@mathieu.bidon.ca
```

**Comment by Fushko** [comment]
Same issue here. Only problem in this essential add-on.

**Comment by otsaloma** [comment]
FWIW, it's fairly easy to hack this in desktop-icons CSS.

```patch
diff -rub desktop-icons@csoriano.orig/desktopGrid.js desktop-icons@csoriano/desktopGrid.js
--- desktop-icons@csoriano.orig/desktopGrid.js	2019-01-05 00:41:26.933528776 +0200
+++ desktop-icons@csoriano/desktopGrid.js	2019-01-05 00:40:53.097574000 +0200
@@ -89,6 +89,7 @@
             y_expand: true,
             can_focus: true,
             opacity: 255,
+            style_class: 'desktop-grid',
         });
         this.actor.add_actor(this._grid);
 
diff -rub desktop-icons@csoriano.orig/stylesheet.css desktop-icons@csoriano/stylesheet.css
--- desktop-icons@csoriano.orig/stylesheet.css	2019-01-05 00:41:19.033539475 +0200
+++ desktop-icons@csoriano/stylesheet.css	2019-01-05 00:44:18.209296655 +0200
@@ -1,3 +1,7 @@
+.desktop-grid {
+    padding-top: 48px;
+}
+
 .file-item {
     padding: 4px;
     border: 1px;
```

**Comment by Nandete84** [comment]
I'm having the same problem but I'm quite new to this world... @otsaloma could you explain where to implement that code, please? Thanks.

**Comment by otsaloma** [comment]
It depends on how you installed the desktop-icons extension, but those files are likely either in `~/.local/share/gnome-shell/extensions/desktop-icons@csoriano` or `/usr/share/gnome-shell/extensions/desktop-icons@csoriano`. After editing you'll need to restart gnome-shell to see if it worked: easiest via Alt+F2, type in "r", press Enter.

**Comment by Nandete84** [comment]
Not working...

I've been really careful in editing both files, but as soon as I reload the shell, the icons in the desktop disappear, the extension (I installed it with a command line, and it is showing through tweaks) in tweaks says that 'there was an error loading this extension'...

Now when I try to load/delete/upgrade the extension (through firefox Gnome shell integration), it says it is enabled by my current mode, so I had to re-install it manually...

**Comment by tuxor1337** [comment]
Solved by #322
